src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,332 @@
+     *
+     * @param jiraStory - new story from Jira
+     */
+    private void createSuitFromJiraStory(JiraStory jiraStory) {
+
+        Suit suit = new Suit();
+        suit.setName(jiraStory.getName());
+        suit.setDescription(jiraStory.getDescription());
+        suit.setJiraKey(jiraStory.getJiraKey());
+        suit.setJiraProjectKey(jiraStory.getJiraProjectKey());
+        suit.setPriority(getPriority(jiraStory.getPriority()));
+        suit.setLastModifiedDate(LocalDateTime.now());
+        suit.setLastJiraSyncDate(suit.getLastModifiedDate());
+        suit = suitDAO.save(suit);
+        Project project = projectDAO.findByJiraKey(jiraStory.getJiraProjectKey());
+        checkNotNull(project);
+
+        project.getSuits().add(suit);
+
+    }
+
+    /**
+     * Creates and associates a new case in BDD to Jira sub story.
+     *
+     * @param jiraSubTask - new sub story from Jira
+     */
+    private void createCaseFromJiraSubTask(JiraSubTask jiraSubTask) {
+        Case caze = new Case();
+        caze.setName(jiraSubTask.getName());
+        caze.setDescription(jiraSubTask.getDescription());
+        caze.setPriority(getPriority(jiraSubTask.getPriority()));
+        caze.setJiraKey(jiraSubTask.getJiraKey());
+        caze.setJiraParentKey(jiraSubTask.getJiraParentKey());
+        caze.setJiraProjectKey(jiraSubTask.getJiraProjectKey());
+        caze.setLastModifiedDate(LocalDateTime.now());
+        caze.setLastJiraSyncDate(caze.getLastModifiedDate());
+
+        caseDAO.save(caze);
+
+        Suit suit = suitDAO.findByJiraKey(jiraSubTask.getJiraParentKey());
+        checkNotNull(suit);
+        suit.getCases().add(caze);
+    }
+
+
+    /**
+     * Updates existent suit in BDD by related Jira story.
+     *
+     * @param jiraStory - related issue from Jira
+     */
+    private void updateSuitFromJiraStory(JiraStory jiraStory) {
+        Suit suit = suitDAO.findByJiraKey(jiraStory.getJiraKey());
+        if (suit == null) {
+            return;
+        }
+        suit.setName(jiraStory.getName());
+        suit.setDescription(jiraStory.getDescription());
+        suit.setPriority(getPriority(jiraStory.getPriority()));
+        suit.setLastJiraSyncDate(LocalDateTime.now());
+
+        suitDAO.save(suit);
+    }
+
+
+    /**
+     * Updates existent case in BDD by related Jira sub story.
+     *
+     * @param jiraSubTask - related issue from Jira
+     */
+    private void updateSubTaskFromJiraSubTask(JiraSubTask jiraSubTask) {
+        Case caze = caseDAO.findByJiraKey(jiraSubTask.getJiraKey());
+        if (caze == null) {
+            return;
+        }
+        caze.setName(jiraSubTask.getName());
+        caze.setDescription(jiraSubTask.getDescription());
+        caze.setPriority(getPriority(jiraSubTask.getPriority()));
+        caze.setLastJiraSyncDate(LocalDateTime.now());
+        caseDAO.save(caze);
+    }
+
+    private Integer getPriority(String priority) {
+        if (priority.equals("Critical")) {
```

Comment:
If priority is NULL you will have NullPointerException, so you need to rewrite conditions and compare string with your priority variable. For example:
```java
 if ("Critical".equals(priority))
```
In this situation, if priority is NULL, you won't have NullPointerException.

Reply:
And all conditions below need to rewrite too.

Reply:
Done.